src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs

```diff
@@ -125,6 +128,36 @@ public SafeX509ChainHandle SafeHandle
             }
         }
 
+        private static SafeX509StoreHandle GetTrustStore(
+            X509ChainTrustMode trustMode,
+            X509Certificate2Collection customTrustStore,
+            SafeX509StackHandle untrusted,
+            SafeX509StackHandle systemTrust)
+        {
+            if (trustMode == X509ChainTrustMode.CustomRootTrust)
+            {
+                SafeX509StackHandle customTrust = Interop.Crypto.NewX509Stack();
+                foreach (X509Certificate2 cert in customTrustStore)
+                {
+                    if (cert.SubjectName.RawData.ContentsEqual(cert.IssuerName.RawData))
+                    {
+                        AddToStackAndUpRef(((OpenSslX509CertificateReader)cert.Pal).SafeHandle, customTrust);
+                    }
+                    else
+                    {
+                        AddToStackAndUpRef(((OpenSslX509CertificateReader)cert.Pal).SafeHandle, untrusted);
+                    }
```

Comment:
This can be simplified to:
```C#
AddToStackAndUpRef(((OpenSslX509CertificateReader)cert.Pal).SafeHandle, 
    cert.SubjectName.RawData.ContentsEqual(cert.IssuerName.RawData) ? customTrust : untrusted);
```
or if you prefer less in a single statement:
```C#
SafeX509StackHandle toAdd = cert.SubjectName.RawData.ContentsEqual(cert.IssuerName.RawData) ? customTrust : untrusted;
AddToStackAndUpRef(((OpenSslX509CertificateReader)cert.Pal).SafeHandle, toAdd);
```
Either way, we avoid the duplicated calls to AddToStackAndUpRef.

Reply:
(Personally, I prefer the latter (saving the `toAdd` local))